Content.Shared/Blocking/SharedBlockingSystem.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Audio;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Item;
+using Content.Shared.Physics;
+using Content.Shared.Popups;
+using Content.Shared.Toggleable;
+using Robust.Shared.Audio;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Collision.Shapes;
+using Robust.Shared.Physics.Dynamics;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Blocking;
+
+public sealed class SharedBlockingSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly FixtureSystem _fixtureSystem = default!;
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SharedBlockingComponent, GettingPickedUpAttemptEvent>(OnPickupAttempt);
+        SubscribeLocalEvent<SharedBlockingComponent, DroppedEvent>(OnDrop);
+
+        SubscribeLocalEvent<SharedBlockingComponent, GetItemActionsEvent>(OnGetActions);
+        SubscribeLocalEvent<SharedBlockingComponent, ToggleActionEvent>(OnToggleAction);
+
+        SubscribeLocalEvent<SharedBlockingUserComponent, DamageModifyEvent>(OnUserDamageModified);
+
+        SubscribeLocalEvent<SharedBlockingComponent, ComponentShutdown>(OnShutdown);
+    }
+
+    private void OnPickupAttempt(EntityUid uid, SharedBlockingComponent component, GettingPickedUpAttemptEvent args)
+    {
```

Comment:
This is not the right event to use as it's raised whenever an attempt is made and not when it's actually picked up.

Reply:
Can you point me to the right event please?